lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -797,6 +802,7 @@ MayaUsdProxyShapeBase::setDependentsDirty(const MPlug& plug, MPlugArray& plugArr
         plug == loadPayloadsAttr ||
         plug == inStageDataAttr) {
         _IncreaseUsdStageVersion();
+        MayaUsdProxyStageInvalidateNotice(*this).Send();
```

Comment:
Send the new stage invalidate notice whenever one of the plugs that affects outStageDataAttr changes.